utils/index.ts

```diff
@@ -215,3 +218,29 @@ export function getImageResizeURL (url: string, { width = 300 }: { width?: numbe
   const { LIKE_CO_STATIC_ENDPOINT } = useRuntimeConfig().public
   return `${LIKE_CO_STATIC_ENDPOINT}/thumbnail/?url=${encodeURIComponent(url)}&width=${width}`
 }
+
+export function fileToArrayBuffer (file: Blob): Promise<string | ArrayBuffer | null> {
+  return new Promise((resolve) => {
+    const reader = new FileReader()
+    reader.onload = () => resolve(reader.result)
+    reader.readAsArrayBuffer(file)
+  })
+}
+
+export async function digestFileSHA256 (buffer: ArrayBuffer) {
+  const hashBuffer = await crypto.subtle.digest('SHA-256', buffer)
+  const hashArray = Array.from(new Uint8Array(hashBuffer))
+  const hashHex = hashArray.map(b => b.toString(16).padStart(2, '0')).join('')
+  return hashHex
+}
+
+export async function calculateIPFSHash (fileBytes: ArrayBuffer) {
+  try {
+    const hash = await sha256.digest(new Uint8Array(fileBytes))
+    const cid = CID.createV1(raw.code, hash)
```

Comment:
Can we use v0? Also please confirm the resulting hash is same as `ipfs-only-hash`

Reply:
> the resulting hash is same as ipfs-only-hash

[🔧 Update calculateIPFSHash importer config](https://github.com/likecoin/nft-book-press/pull/290/commits/49a0ccf69721df4faa3c08853e16a8f3202b39a4)